spec/support/macros.rb

```diff
@@ -0,0 +1,15 @@
+def set_current_user(user=nil)
+  @user = user || Fabricate(:user)
+  controller.sign_in(@user)
+end
+
+def sign_in_user(new_user=nil)
```

Comment:
This should be more like this:

```ruby
def sign_in_user(user = nil)
  user ||= Fabricate(:user)
  visit sign_in_path
  fill_in "Email Address", with: "#{user.email}"
  fill_in "Password", with: "#{user.password}"
  click_button "Sign in"
end
```
You can use your knowledge about memoization. Also the last 3 expectations does not really belong to the macro, if so make a 1 feature spec but this gets carried to everywhere you use `sign_in_user` method. Also spaces around operators - always.